tests/std/tests/P1208R6_source_location/test.cpp

```diff
@@ -102,10 +110,19 @@ constexpr void sloc_constructor_test() {
     } else
 #endif // ^^^ workaround ^^^
     {
+#ifdef __EDG__
+        assert(x.loc.function_name() == "void sloc_constructor_test()"sv);
+#else // ^^^ EDG / Other vvv
         assert(x.loc.function_name() == "void __cdecl sloc_constructor_test(void)"sv);
+#endif // ^^^ Other ^^^
     }
 #else // ^^^ detailed / basic vvv
-    assert(x.loc.function_name() == "sloc_constructor_test"sv);
+#if !defined(__clang__) && !defined(__EDG__) // TRANSITION, VSO-1285783
+    if (is_constant_evaluated()) {
+        assert(x.loc.function_name() == "main"sv);
+    } else
+#endif // ^^^ workaround ^^^
+        assert(x.loc.function_name() == "sloc_constructor_test"sv);

```

Comment:
This sometimes-`else` branch should be fully braced:
```suggestion
    {
        assert(x.loc.function_name() == "sloc_constructor_test"sv);
    }
```